<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="This package contains common analysis, processing and plotting 
    functions with an attempt to streamline single-cell genomics analysis 
    using the Seurat package.">
<title>Packaging common Seurat analysis tasks • SeuratPipe</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Packaging common Seurat analysis tasks">
<meta property="og:description" content="This package contains common analysis, processing and plotting 
    functions with an attempt to streamline single-cell genomics analysis 
    using the Seurat package.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">SeuratPipe</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/website/seurat_pipeline.html">SeuratPipe - Streamlining Seurat analysis</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://andreaskapou.github.io" aria-label="Personal website">Andreas C. Kapourani</a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="seuratpipe">SeuratPipe<a class="anchor" aria-label="anchor" href="#seuratpipe"></a>
</h1></div>
<p>The aim of the <code>SeuratPipe</code> package is to remove technical details from the user, enabling wet lab researchers to have an ‘initial look’ at the data themselves.</p>
<p>The <strong>vignette</strong> demonstrating the use of <code>SeuratPipe</code> for scRNA-seq analysis can be found on Rpubs <a href="https://rpubs.com/cakapourani/seuratpipe-analysis" class="external-link">https://rpubs.com/cakapourani/seuratpipe-analysis</a> and the <strong>data</strong> can be downloaded from this Github repository <a href="https://github.com/andreaskapou/SeuratPipe_tutorial" class="external-link">https://github.com/andreaskapou/SeuratPipe_tutorial</a>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("remotes")</span>

<span class="co"># To install the stable release</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"andreaskapou/SeuratPipe@release/v1.0.0"</span><span class="op">)</span>

<span class="co"># To install the latest source version (not recommended)</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"andreaskapou/SeuratPipe"</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="scrublet-python-installation">Scrublet Python installation<a class="anchor" aria-label="anchor" href="#scrublet-python-installation"></a>
</h3>
<p>Scrublet is a method for doublet detection, Github page: <a href="https://github.com/swolock/scrublet" class="external-link">https://github.com/swolock/scrublet</a>.</p>
<p>The method is implemented in Python, hence when installing <code>SeuratPipe</code>, the Scrublet package cannot be installed automatically. We can try to install and run Scrublet within R using the <code>reticulate</code> package. I wrote a helper function to ease installation of Scrublet. In R type:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">SeuratPipe</span><span class="fu">::</span><span class="fu"><a href="reference/install_scrublet.html">install_scrublet</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># type ?install_scrublet for more details</span></code></pre></div>
<p>If you have any issues with installation, you can still use the SeuratPipe package, although you have to set <code>use_scrublet = FALSE</code>, i.e. do not perform any doublet detection in your analysis.</p>
</div>
</div>
<div class="section level2">
<h2 id="qc-analysis-pipeline">QC analysis pipeline<a class="anchor" aria-label="anchor" href="#qc-analysis-pipeline"></a>
</h2>
<p><code>SeuratPipe</code> assumes the user will have a <strong>sample metadata</strong> file/data frame which contains all the metadata for each scRNA-seq sample. It also requires the following column names to be present: <code>sample</code>, <code>donor</code>, <code>path</code>, <code>condition</code> and <code>pass_qc</code> (logical <code>TRUE</code> or <code>FALSE</code>).</p>
<p>We can then perform all QC pipeline steps using a single wrapper function <code>run_qc_pipeline</code> as shown below. This function will also generate all the required QC plots.</p>
<pre class="{r"><code><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/run_qc_pipeline.html">run_qc_pipeline</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span>
<span class="co"># type ?run_qc_pipeline for argument details</span></code></pre>
</div>
<div class="section level2">
<h2 id="clustering-analysis-pipeline">Clustering analysis pipeline<a class="anchor" aria-label="anchor" href="#clustering-analysis-pipeline"></a>
</h2>
<p>Here we show how to perform the most common analysis pipeline in Seurat for cell type identification. The wrapper function performs the following steps:</p>
<ol style="list-style-type: decimal">
<li>Data processing, e.g. Log normalisation and scaling.</li>
<li>Run PCA.</li>
<li>Perform UMAP and clustering.</li>
<li>Generate plots / outputs in corresponding directories.</li>
</ol>
<pre class="{r"><code><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/run_cluster_pipeline.html">run_cluster_pipeline</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span>
<span class="co"># type ?run_cluster_pipeline for argument details</span></code></pre>
</div>
<div class="section level2">
<h2 id="harmony-integration-pipeline">Harmony integration pipeline<a class="anchor" aria-label="anchor" href="#harmony-integration-pipeline"></a>
</h2>
<p>The Harmony integration pipeline is almost identical to <code>run_cluster_pipeline</code>, with the main difference of the function accepting a list of Seurat objects (each containing information for a single sample) to perform data integration.</p>
<pre class="{r"><code><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/run_harmony_pipeline.html">run_harmony_pipeline</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span>
<span class="co"># type ?run_harmony_pipeline for argument details</span></code></pre>
</div>
</div>
<div class="section level1">
<h1 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h1>
<p>I would like to thank John Wilson-Kanamori, Jordan Portman and Nick Younger. The <code>SeuratPipe</code> package has adapted many of the analyses pipelines that were developed by the computational biology group in Neil Henderson’s lab at the University of Edinburgh.</p>
<p>This package was supported by funding from the University of Edinburgh and Medical Research Council (core grant to the MRC Institute of Genetics and Cancer) for the Cross-Disciplinary Fellowship (XDF) programme.</p>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> | file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SeuratPipe</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Andreas C. Kapourani <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2303-1953" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Andreas C. Kapourani.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
