<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function implements all the analysis steps for performing
clustering on a Seurat object. These include, 1. finding neighbours in lower
dimensional space (defined in 'cluster_reduction' parameter) 2. obtaining
clusters, 3. identifying marker genes (NOTE: to speed up re-analysis it
first checks if file with marker genes is already present, if yes reads the
file instead of calling FinaAllMarkers) and 4. generating plots, which
include heatmap with (scaled) expression of marker genes in each cluster,
marker gene expression on feature plots (e.g. UMAP space, defined in
plot_reduction'  parameter), dot / feature plots with
pre-computed module scores on each cluster (assumes we have first run
'module_score_analysis' function). This step could be useful for
lineage annotation."><title>Common clustering analysis steps — cluster_analysis • SeuratPipe</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Common clustering analysis steps — cluster_analysis"><meta property="og:description" content="This function implements all the analysis steps for performing
clustering on a Seurat object. These include, 1. finding neighbours in lower
dimensional space (defined in 'cluster_reduction' parameter) 2. obtaining
clusters, 3. identifying marker genes (NOTE: to speed up re-analysis it
first checks if file with marker genes is already present, if yes reads the
file instead of calling FinaAllMarkers) and 4. generating plots, which
include heatmap with (scaled) expression of marker genes in each cluster,
marker gene expression on feature plots (e.g. UMAP space, defined in
plot_reduction'  parameter), dot / feature plots with
pre-computed module scores on each cluster (assumes we have first run
'module_score_analysis' function). This step could be useful for
lineage annotation."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SeuratPipe</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/website/SeuratPipe.html">SeuratPipe - Streamlining Seurat analysis</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://andreaskapou.github.io" aria-label="Personal website">Andreas C. Kapourani</a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Common clustering analysis steps</h1>
      
      <div class="d-none name"><code>cluster_analysis.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function implements all the analysis steps for performing
clustering on a Seurat object. These include, 1. finding neighbours in lower
dimensional space (defined in 'cluster_reduction' parameter) 2. obtaining
clusters, 3. identifying marker genes (NOTE: to speed up re-analysis it
first checks if file with marker genes is already present, if yes reads the
file instead of calling FinaAllMarkers) and 4. generating plots, which
include heatmap with (scaled) expression of marker genes in each cluster,
marker gene expression on feature plots (e.g. UMAP space, defined in
plot_reduction'  parameter), dot / feature plots with
pre-computed module scores on each cluster (assumes we have first run
'module_score_analysis' function). This step could be useful for
lineage annotation.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">cluster_analysis</span><span class="op">(</span>
  <span class="va">seu</span>,
  dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,
  res <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.1</span>, by <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>,
  logfc.threshold <span class="op">=</span> <span class="fl">0.5</span>,
  min.pct <span class="op">=</span> <span class="fl">0.25</span>,
  only.pos <span class="op">=</span> <span class="cn">TRUE</span>,
  topn_genes <span class="op">=</span> <span class="fl">10</span>,
  diff_cluster_pct <span class="op">=</span> <span class="fl">0.1</span>,
  pval_adj <span class="op">=</span> <span class="fl">0.05</span>,
  plot_dir <span class="op">=</span> <span class="cn">NULL</span>,
  plot_cluster_markers <span class="op">=</span> <span class="cn">TRUE</span>,
  modules_group <span class="op">=</span> <span class="cn">NULL</span>,
  cluster_reduction <span class="op">=</span> <span class="st">"pca"</span>,
  plot_reduction <span class="op">=</span> <span class="st">"umap"</span>,
  max.cutoff <span class="op">=</span> <span class="st">"q98"</span>,
  min.cutoff <span class="op">=</span> <span class="cn">NA</span>,
  seed <span class="op">=</span> <span class="fl">1</span>,
  force_reanalysis <span class="op">=</span> <span class="cn">TRUE</span>,
  label <span class="op">=</span> <span class="cn">TRUE</span>,
  label.size <span class="op">=</span> <span class="fl">8</span>,
  legend.position <span class="op">=</span> <span class="st">"right"</span>,
  pt.size <span class="op">=</span> <span class="fl">1.4</span>,
  cont_col_pal <span class="op">=</span> <span class="cn">NULL</span>,
  discrete_col_pal <span class="op">=</span> <span class="cn">NULL</span>,
  fig.res <span class="op">=</span> <span class="fl">200</span>,
  heatmap_downsample_cols <span class="op">=</span> <span class="cn">NULL</span>,
  cont_alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.9</span><span class="op">)</span>,
  discrete_alpha <span class="op">=</span> <span class="fl">0.9</span>,
  pt.size.factor <span class="op">=</span> <span class="fl">1.1</span>,
  spatial_col_pal <span class="op">=</span> <span class="st">"inferno"</span>,
  crop <span class="op">=</span> <span class="cn">FALSE</span>,
  plot_spatial_markers <span class="op">=</span> <span class="cn">FALSE</span>,
  spatial_legend_position <span class="op">=</span> <span class="st">"top"</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>seu</dt>
<dd><p>Seurat object (required).</p></dd>
<dt>dims</dt>
<dd><p>Vector denoting dimensions to use for nearest neighnors and
clustering (from 'cluster_reduction' parameter below).</p></dd>
<dt>res</dt>
<dd><p>Vector with clustering resolutions (e.g. seq(0.1, 0.6, by = 0.1)).</p></dd>
<dt>logfc.threshold</dt>
<dd><p>Limit testing to genes which show, on average, at
least X-fold difference (log-scale) between the two groups of cells.</p></dd>
<dt>min.pct</dt>
<dd><p>Only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations.</p></dd>
<dt>only.pos</dt>
<dd><p>Only return positive markers (TRUE by default).</p></dd>
<dt>topn_genes</dt>
<dd><p>Top cluster marker genes to use for plot (in heatmap and
feature plots), default is 10.</p></dd>
<dt>diff_cluster_pct</dt>
<dd><p>Retain marker genes per cluster if their
<code>pct.1 - pct.2 &gt; diff_cluster_pct</code>, i.e. they show cluster
specific expression. Set to -Inf, to ignore this additional filtering.</p></dd>
<dt>pval_adj</dt>
<dd><p>Adjusted p-value threshold to consider marker genes per cluster.</p></dd>
<dt>plot_dir</dt>
<dd><p>Directory to save generated plots. If NULL, plots are
not saved.</p></dd>
<dt>plot_cluster_markers</dt>
<dd><p>Logical, whether to create feature plots with
'topn_genes' cluster markers. Added mostly to reduce number of files (and size)
in analysis folders. Default is TRUE.</p></dd>
<dt>modules_group</dt>
<dd><p>Group of modules (named list of lists) storing features
(e.g. genes) to compute module score for each identified cluster. This step
can be useful for annotating the different clusters by saving dot
plots for each group. Assumes that we already have computed the modules e.g.
by calling the 'module_score_analysis' function. If 'plot_dir' is NULL,
no plots will be generated.</p></dd>
<dt>cluster_reduction</dt>
<dd><p>Dimensionality reduction to use for performing
clustering. Default is 'pca', should be set to 'harmony' if we perform data
integration.</p></dd>
<dt>plot_reduction</dt>
<dd><p>Dimensionality reduction to use for plotting
functions. Default is 'umap'.</p></dd>
<dt>max.cutoff</dt>
<dd><p>Vector of maximum cutoff values for each feature,
may specify quantile in the form of 'q##' where '##' is the quantile
(eg, 'q1', 'q10').</p></dd>
<dt>min.cutoff</dt>
<dd><p>Vector of minimum cutoff values for each feature,
may specify quantile in the form of 'q##' where '##' is the quantile
(eg, 'q1', 'q10').</p></dd>
<dt>seed</dt>
<dd><p>Set a random seed, for reproducibility.</p></dd>
<dt>force_reanalysis</dt>
<dd><p>Logical, if cluster marker genes file
exists and force_reanalysis = FALSE, run identification of cluster markers.
Otherwise, read cluster markers from file. Added for computing time
efficiency purposes.</p></dd>
<dt>label</dt>
<dd><p>Whether to label the clusters in 'plot_reduction' space.</p></dd>
<dt>label.size</dt>
<dd><p>Sets size of labels.</p></dd>
<dt>legend.position</dt>
<dd><p>Position of legend, default "right" (set to "none"
for clean plot).</p></dd>
<dt>pt.size</dt>
<dd><p>Adjust point size for plotting.</p></dd>
<dt>cont_col_pal</dt>
<dd><p>Continuous colour palette to use, default "RdYlBu".</p></dd>
<dt>discrete_col_pal</dt>
<dd><p>Discrete colour palette to use, default is Hue palette
(hue_pal) from 'scales' package.</p></dd>
<dt>fig.res</dt>
<dd><p>Figure resolution in ppi (see 'png' function).</p></dd>
<dt>heatmap_downsample_cols</dt>
<dd><p>If numberic, it will downsamples the columns of
the heatmap plot, so a big specific cluster doesn't dominate the heatmap.</p></dd>
<dt>cont_alpha</dt>
<dd><p>Controls opacity of spots. Provide as a vector specifying the
min and max range of values (between 0 and 1).</p></dd>
<dt>discrete_alpha</dt>
<dd><p>Controls opacity of spots. Provide a single alpha value.</p></dd>
<dt>pt.size.factor</dt>
<dd><p>Scale the size of the spots.</p></dd>
<dt>spatial_col_pal</dt>
<dd><p>Continuous colour palette to use from viridis package to
colour spots on tissue, default "inferno".</p></dd>
<dt>crop</dt>
<dd><p>Crop the plot in to focus on spots that passed QC. Set to FALSE to
show entire background image.</p></dd>
<dt>plot_spatial_markers</dt>
<dd><p>Logical, whether to create spatial feature plots
with expression of individual genes.</p></dd>
<dt>spatial_legend_position</dt>
<dd><p>Position of legend for spatial plots,
default "top" (set to "none" for clean plot).</p></dd>
<dt>...</dt>
<dd><p>Additional named parameters passed to Seurat
analysis and plotting functions, such as FindClusters, FindAllMarkers,
DimPlot and FeaturePlot.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Updated Seurat object clustered cells</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>C.A.Kapourani <a href="mailto:C.A.Kapourani@ed.ac.uk">C.A.Kapourani@ed.ac.uk</a></p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Andreas C. Kapourani.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

