<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function implements all the analysis steps for perfoming
QC. These include: 1. reading all sample information from metadata object/file
and generating one Seurat object per sample. 2. Performs SoupX (ambient
RNA removal) and Scrublet (doublet detection) if user defines the
corresponding parameters. 3. Filter Seurat object according to QC
criteria 4. Generate correspond QC plots."><title>QC pipeline — run_qc_pipeline • SeuratPipe</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="QC pipeline — run_qc_pipeline"><meta property="og:description" content="This function implements all the analysis steps for perfoming
QC. These include: 1. reading all sample information from metadata object/file
and generating one Seurat object per sample. 2. Performs SoupX (ambient
RNA removal) and Scrublet (doublet detection) if user defines the
corresponding parameters. 3. Filter Seurat object according to QC
criteria 4. Generate correspond QC plots."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SeuratPipe</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/website/SeuratPipe.html">SeuratPipe for scRNA-seq analysis</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://andreaskapou.github.io" aria-label="Personal website">Andreas C. Kapourani</a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>QC pipeline</h1>
      
      <div class="d-none name"><code>run_qc_pipeline.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function implements all the analysis steps for perfoming
QC. These include: 1. reading all sample information from metadata object/file
and generating one Seurat object per sample. 2. Performs SoupX (ambient
RNA removal) and Scrublet (doublet detection) if user defines the
corresponding parameters. 3. Filter Seurat object according to QC
criteria 4. Generate correspond QC plots.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">run_qc_pipeline</span><span class="op">(</span>
  <span class="va">data_dir</span>,
  <span class="va">sample_meta</span>,
  sample_meta_filename <span class="op">=</span> <span class="cn">NULL</span>,
  nfeat_thresh <span class="op">=</span> <span class="fl">500</span>,
  mito_thresh <span class="op">=</span> <span class="fl">5</span>,
  meta_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"donor"</span>, <span class="st">"condition"</span>, <span class="st">"pass_qc"</span><span class="op">)</span>,
  out_dir <span class="op">=</span> <span class="cn">NULL</span>,
  qc_to_plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"nFeature_RNA"</span>, <span class="st">"nCount_RNA"</span>, <span class="st">"percent.mito"</span><span class="op">)</span>,
  use_scrublet <span class="op">=</span> <span class="cn">TRUE</span>,
  use_soupx <span class="op">=</span> <span class="cn">FALSE</span>,
  tenx_dir <span class="op">=</span> <span class="st">"premrna_outs"</span>,
  tenx_counts_dir <span class="op">=</span> <span class="st">"filtered_feature_bc_matrix"</span>,
  obj_filename <span class="op">=</span> <span class="st">"seu_qc"</span>,
  expected_doublet_rate <span class="op">=</span> <span class="fl">0.06</span>,
  force_reanalysis <span class="op">=</span> <span class="cn">TRUE</span>,
  min.cells <span class="op">=</span> <span class="fl">10</span>,
  min.features <span class="op">=</span> <span class="fl">100</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>data_dir</dt>
<dd><p>Parent directory where all sample 10x files are stored.
Think of it as project directory.</p></dd>
<dt>sample_meta</dt>
<dd><p>Sample metadata information in a Data.frame like object.
Columns should at least contain 'sample', 'donor', 'condition' and 'pass_qc'.</p></dd>
<dt>sample_meta_filename</dt>
<dd><p>Filename of sample metadata information, same as 'meta'
parameter above. User should provide one of 'meta' or 'meta_filename'.</p></dd>
<dt>nfeat_thresh</dt>
<dd><p>Filter cells that have less than 'nfeat_thresh' counts
expressed.</p></dd>
<dt>mito_thresh</dt>
<dd><p>Filter cells with more than 'mito_thresh'% counts.</p></dd>
<dt>meta_colnames</dt>
<dd><p>Sample metadata column names to store in Seurat metadata.</p></dd>
<dt>out_dir</dt>
<dd><p>Output directory for storing analysis results.</p></dd>
<dt>qc_to_plot</dt>
<dd><p>Vector of features in metadata to plot.</p></dd>
<dt>use_scrublet</dt>
<dd><p>Logical, wether to use Scrublet for doublet detection.</p></dd>
<dt>use_soupx</dt>
<dd><p>Logical, wether to use SoupX for ambient RNA removal.</p></dd>
<dt>tenx_dir</dt>
<dd><p>Name of 10x base directory, e.g. with outputs after running
cellranger. Default 'premrna_outs', i.e. assumes single-nuclei RNA-seq.</p></dd>
<dt>tenx_counts_dir</dt>
<dd><p>Name of 10x directory where count matrices are stored.
Default 'filtered_feature_bc_matrix'</p></dd>
<dt>obj_filename</dt>
<dd><p>Filename of the stored Seurat object, default 'seu_qc'.</p></dd>
<dt>expected_doublet_rate</dt>
<dd><p>The expected fraction of transcriptomes that
are doublets, typically 0.05 - 0.1</p></dd>
<dt>force_reanalysis</dt>
<dd><p>Logical, if intermediate file 'seu_preqc.rds' (with
created Seurat object) exists and force_reanalysis = FALSE, read object
instead of re-running whole analysis with soupX. Added for computing time
efficiency purposes and intermediate object will be created only when
'use_soupx = TRUE'.</p></dd>
<dt>min.cells</dt>
<dd><p>Include features/genes detected in at least this many cells.</p></dd>
<dt>min.features</dt>
<dd><p>Include cells where at least this many features/genes
are detected.</p></dd>
<dt>...</dt>
<dd><p>Additional named parameters passed to Seurat, Scrublet or SoupX.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>List of Seurat objects as the length of the number of samples in
the sample metadata file. If a single sample, return a Seurat object
instead of a list.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>C.A.Kapourani <a href="mailto:C.A.Kapourani@ed.ac.uk">C.A.Kapourani@ed.ac.uk</a></p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Andreas C. Kapourani.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

